components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/jwks/JwksEndpoint.java

```diff
@@ -65,50 +68,84 @@ public String jwks() {
                 if (isInvalidTenantId(tenantId)) {
                     String errorMessage = "Invalid Tenant: " + tenantDomain;
                     return logAndReturnError(errorMessage, null);
                 }
                 KeyStoreManager keyStoreManager = KeyStoreManager.getInstance(tenantId);
-                KeyStore keyStore = keyStoreManager.getKeyStore(generateKSNameFromDomainName(tenantDomain));
-                // Get certificate of public key
-                publicKey = getRsaPublicKey(tenantDomain, keyStore);
+                keystore = keyStoreManager.getKeyStore(generateKSNameFromDomainName(tenantDomain));
+            }
+            Enumeration enumeration = keystore.aliases();
+            while (enumeration.hasMoreElements()) {
+                String alias = (String) enumeration.nextElement();
+                if (keystore.isKeyEntry(alias)) {
+                    Certificate cert = keystore.getCertificate(alias);
+                    certKeyPair.put(alias,cert);
+                }
             }
-            return buildResponse(tenantDomain, tenantId, publicKey);
+            return buildResponse(certKeyPair);
         } catch (Exception e) {
-            String errorMessage = "Error while generating the keyset for " + tenantDomain + " tenant domain.";
+            String errorMessage = "Error while generating the keyset for tenant domain: " + tenantDomain;
             return logAndReturnError(errorMessage, e);
         }
     }
 
-    private String buildResponse(String tenantDomain, int tenantId, RSAPublicKey publicKey)
+    private String buildResponse(HashMap<String, Certificate> certificates)
             throws IdentityOAuth2Exception, ParseException {
+
         JSONArray jwksArray = new JSONArray();
         JSONObject jwksJson = new JSONObject();
-
-        RSAKey.Builder jwk = new RSAKey.Builder(publicKey);
-        jwk.keyID(OAuth2Util.getThumbPrint(tenantDomain, tenantId));
-        jwk.algorithm(JWSAlgorithm.RS256);
-        jwk.keyUse(KeyUse.parse(KEY_USE));
-        jwksArray.put(jwk.build().toJSONObject());
+        OAuthServerConfiguration config = OAuthServerConfiguration.getInstance();
+        JWSAlgorithm accessTokenSignAlgorithm =
+                OAuth2Util.mapSignatureAlgorithmForJWSAlgorithm(config.getIdTokenSignatureAlgorithm());
+        for (Map.Entry certKeyPair : certificates.entrySet()) {
+            Certificate cert = (Certificate) certKeyPair.getValue();
+            String alias = (String) certKeyPair.getKey();
+            RSAPublicKey publicKey = (RSAPublicKey) cert.getPublicKey();
+            RSAKey.Builder jwk = new RSAKey.Builder(publicKey);
+            jwk.keyID(OAuth2Util.getThumbPrint(cert, alias));
+            jwk.algorithm(accessTokenSignAlgorithm);
+            jwk.keyUse(KeyUse.parse(KEY_USE));
+            jwksArray.put(jwk.build().toJSONObject());
+        }
+        //read all the signature algorithms from identity.xml and put them in a list if there are different
```

Comment:
This comment does not make sense.

Consider comments as sentences. 
- Leave a space after //
- Start with a capital letter
- End with the full stop